Wizard_chess.py

```diff
@@ -1,728 +1,308 @@
-import pickle
 import threading
+import chess
+
 import RPi.GPIO as GPIO
 from motor_class import Motor
 
-motor_x = Motor(17,18,27,22)
-motor_y = Motor(4,25,24,23)
-motor_x.init()
-motor_y.init()
-
-GPIO.setup(7, GPIO.OUT)
-GPIO.output(7,1)
-
-motor_movement_dictionary_from_a1 = {
-"00":[0,0],
-"10":[0,0.65],
-"20":[0,1.3],
-"30":[0,1.95],
-"40":[0,2.6],
-"50":[0,3.25],
-"60":[0,3.9],
-"70":[0,4.55],
-"01":[0.65,],
-"11":[0.65,0.65],
-"21":[0.65,1.3],
-"31":[0.65,1.95],
-"41":[0.65,2.6],
-"51":[0.65,3.25],
-"61":[0.65,3.9],
-"71":[0.65,4.55],
-"02":[1.3,0],
-"12":[1.3,0.65],
-"22":[1.3,1.3],
-"32":[1.3,1.95],
-"42":[1.3,2.6],
-"52":[1.3,3.25],
-"62":[1.3,3.9],
-"72":[1.3,4.55],
-"03":[1.95,0],
-"13":[1.95,0.65],
-"23":[1.95,1.3],
-"33":[1.95,1.95],
-"43":[1.95,2.6],
-"53":[1.95,3.25],
-"63":[1.95,3.9],
-"73":[1.95,4.55],
-"04":[2.6,0],
-"14":[2.6,0.65],
-"24":[2.6,1.3],
-"34":[2.6,1.95],
-"44":[2.6,2.6],
-"54":[2.6,3.25],
-"64":[2.6,3.9],
-"74":[2.6,4.55],
-"05":[3.25,0],
-"15":[3.25,0.65],
-"25":[3.25,1.3],
-"35":[3.25,1.95],
-"45":[3.25,2.6],
-"55":[3.25,3.25],
-"65":[3.25,3.9],
-"75":[3.25,4.55],
-"06":[3.9,0],
-"16":[3.9,0.65],
-"26":[3.9,1.3],
-"36":[3.9,1.95],
-"46":[3.9,2.6],
-"56":[3.9,3.25],
-"66":[3.9,3.9],
-"76":[3.9,4.55],
-"07":[4.55,0],
-"17":[4.55,0.65],
-"27":[4.55,1.3],
-"37":[4.55,1.95],
-"47":[4.55,2.6],
-"57":[4.55,3.25],
-"67":[4.55,3.9],
-"77":[4.55,4.55],
-}
-
-class piece():
-    pieces = {'white':{},'black':{}}
-    pieces_taken = {'white':{},'black':{}}
-    def __init__(self,piece_type,colour):
-        self.piece_type=piece_type
-        self.colour=colour
-        self.no_of_moves = 0
-        self.position = '00'
-        self.moves_list = []
-
-        if not self.piece_type in piece.pieces[self.colour]:
-            piece.pieces[self.colour][self.piece_type] = [self]
+
+class Magnet:
+    """
+    Simple wrapper around GPIO calls, just to make it more clear what's g oin
+    """
+
+    @staticmethod
+    def initialize():
+        """
+        Sets up the magnet
+        :return: none
+        """
+        GPIO.setup(7, GPIO.OUT)
+
+    @staticmethod
+    def grab():
+        """
+        Grab a piece with the magnet
+        :return: none
+        """
+        GPIO.output(7, 0)

```

Comment:
I think these values are correct (i.e. 0 means move the magnet up to grab a piece and 1 means move the magnet down to release a piece), but please check :)